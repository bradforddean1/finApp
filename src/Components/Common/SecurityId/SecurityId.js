import React from "react";
import PropTypes from "prop-types";
import "./SecurityId.css";

/**
 * Displays secuity identifying information, Logo, Name and Ticker Symbol
 * @component
 */
function SecurityId(props) {
	return (
		<div className="SecurityId card-line">
			{!!props.logo && <img src={props.logo} alt="logo" />}
			<h2>
				{props.name} <span>({props.ticker})</span>
			</h2>
		</div>
	);
}

SecurityId.propTypes = {
	logo: PropTypes.string,
	name: PropTypes.string.isRequired,
	ticker: PropTypes.string.isRequired,
};

SecurityId.defaultProps = {
	logo: null,
};

export default SecurityId;
