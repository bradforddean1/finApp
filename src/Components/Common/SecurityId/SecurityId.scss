.SecurityId {
	display: flex;
	align-items: center;
}

.SecurityId img {
	margin-right: 0.83em;
	background-color: #fff;
	border: 0px solid;
	border-radius: 45px;
	height: 1.3333rem;
	width: 1.3333rem;
}
