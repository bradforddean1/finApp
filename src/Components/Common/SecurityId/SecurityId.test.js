import React from "react";
import ReactDOM from "react-dom";
import SecurityId from "./SecurityId";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("SecurityId Component", () => {
	it("Renders", () => {
		const div = document.createElement("div");
		ReactDOM.render(<SecurityId />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it("Renders the Default State", () => {
		const wrapper = shallow(<SecurityId />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	const props = {
		logo: "https://test.logo.com",
		name: "XYZ, corp",
		ticker: "ZYZ",
	};

	it("Renders given props", () => {
		const wrapper = shallow(<SecurityId {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
