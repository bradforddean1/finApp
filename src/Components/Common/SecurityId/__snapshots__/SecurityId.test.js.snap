// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecurityId Component Renders given props 1`] = `
<div
  className="SecurityId card-line"
>
  <img
    alt="logo"
    src="https://test.logo.com"
  />
  <h2>
    XYZ, corp
     
    <span>
      (
      ZYZ
      )
    </span>
  </h2>
</div>
`;

exports[`SecurityId Component Renders the Default State 1`] = `
<div
  className="SecurityId card-line"
>
  <h2>
     
    <span>
      (
      )
    </span>
  </h2>
</div>
`;
